src/easydiffraction/experiments/experiments.py

```diff
@@ -25,83 +28,70 @@ def __init__(self) -> None:
-            )
-        else:
-            raise ValueError('Provide either experiment, type parameters, cif_path, cif_str, or data_path')
+        self._add_prebuilt_experiment(experiment)
 
-    @enforce_type
-    def _add_prebuilt_experiment(self, experiment: BaseExperiment):
-        self._experiments[experiment.name] = experiment
+    def add_from_cif_path(self, cif_path: str):
+        """
+        Add a new experiment from a CIF file path.
+        """
+        experiment = Experiment(cif_path=cif_path)
+        self._add_prebuilt_experiment(experiment)
 
-    def _add_from_cif_path(self, cif_path: str) -> None:
-        print('Loading Experiment from CIF path...')
-        raise NotImplementedError('CIF loading not implemented.')
+    def add_from_cif_str(self, cif_str: str):
+        """
+        Add a new experiment from CIF file content (string).
+        """
+        experiment = Experiment(cif_str=cif_str)
+        self._add_prebuilt_experiment(experiment)
 
-    def _add_from_cif_string(self, cif_str: str) -> None:
-        print('Loading Experiment from CIF string...')
-        raise NotImplementedError('CIF loading not implemented.')
+    def add_from_data_path(
+        self,
+        name: str,
+        data_path: str,
+        sample_form: str = DEFAULT_SAMPLE_FORM,
+        beam_mode: str = DEFAULT_BEAM_MODE,
+        radiation_probe: str = DEFAULT_RADIATION_PROBE,
+        scattering_type: str = DEFAULT_SCATTERING_TYPE,
+    ):
+        """
+        Add a new experiment from a data file path.
+        """
+        experiment = Experiment(
+            name=name,
+            data_path=data_path,
+            sample_form=sample_form,
+            beam_mode=beam_mode,
+            radiation_probe=radiation_probe,
+            scattering_type=scattering_type,
+        )
+        self._add_prebuilt_experiment(experiment)
 
-    def _add_from_data_path(
+    def add_without_data(
         self,
-        name,
-        sample_form,
-        beam_mode,
-        radiation_probe,
-        scattering_type,
-        data_path,
+        name: str,
+        sample_form: str = DEFAULT_SAMPLE_FORM,
+        beam_mode: str = DEFAULT_BEAM_MODE,
+        radiation_probe: str = DEFAULT_RADIATION_PROBE,
+        scattering_type: str = DEFAULT_SCATTERING_TYPE,
     ):
         """
-        Load an experiment from raw data ASCII file.
+        Add a new experiment without any data file.
         """
-        print(paragraph('Loading measured data from ASCII file'))
-        print(os.path.abspath(data_path))
-        experiment = ExperimentFactory.create(
+        experiment = Experiment(
             name=name,
             sample_form=sample_form,
             beam_mode=beam_mode,
             radiation_probe=radiation_probe,
             scattering_type=scattering_type,
         )
-        experiment._load_ascii_data_to_experiment(data_path)
+        self._add_prebuilt_experiment(experiment)
+
+    @enforce_type
```

Comment:
I like type enforcement here. It would probably be prudent to try to use `@enforce_type` in other places as well, especially in setters, so we don't have to perform explicit type checks

Reply:
Agreed. I’ve created issue #87 to track this.